google/cloud/pubsub/internal/tracing_message_batch_test.cc

```diff
@@ -130,11 +130,43 @@ TEST(TracingMessageBatch, Flush) {
                                      "messaging.pubsub.message.link", 0)))))));
 }
 
+TEST(TracingMessageBatch, FlushOnlyIncludeSampledLink) {
+  namespace sc = ::opentelemetry::trace::SemanticConventions;
+  // Create span before the span catcher so it is not sampled.
+  auto unsampled_span = MakeSpan("test skipped span");
+  auto span_catcher = InstallSpanCatcher();
+  auto message_span = MakeSpan("test span");
+  auto mock = std::make_unique<pubsub_testing::MockMessageBatch>();
+  EXPECT_CALL(*mock, SaveMessage(_)).Times(2);
+  EXPECT_CALL(*mock, Flush).WillOnce([] {
+    EXPECT_TRUE(ThereIsAnActiveSpan());
+    EXPECT_TRUE(OTelContextCaptured());
+    return [](auto) { EXPECT_FALSE(OTelContextCaptured()); };
+  });
+  auto message_batch = MakeTracingMessageBatch(std::move(mock));
+  auto initial_spans = {message_span, unsampled_span};
+  SaveMessages(initial_spans, message_batch);
+
+  auto end_spans = message_batch->Flush();
+  end_spans(make_ready_future());
+
+  auto spans = span_catcher->GetSpans();
+  EXPECT_THAT(spans,
+              Contains(AllOf(SpanHasInstrumentationScope(), SpanKindIsClient(),
+                             SpanNamed("BatchSink::AsyncPublish"),
+                             SpanHasAttributes(OTelAttribute<std::int64_t>(
+                                 sc::kMessagingBatchMessageCount, 2)),
+                             SpanHasLinks(AllOf(
```

Comment:
I think to verify that `unsampled_span` is not linked to, we want `SpanLinksAre`.

Reply:
done

Reply:
I was using the kMessagingBatchMessageCount also kind of verified it....

what you suggested is much better.

Thanks!